stko/optimizers/gulp.py

```diff
@@ -520,52 +557,131 @@ def _write_gulp_file(self, mol, metal_atoms, in_file, output_xyz):
 
         type_translator = self._type_translator()
 
+        top_line = 'opti '
+
         if self._conjugate_gradient:
-            top_line = (
-                'opti conj unit conv noautobond fix molmec cartesian\n'
-            )
+            top_line += 'conj unit '
+
+        if self._cell is not None:
+            # Constant pressure.
+            top_line += 'conp '
+            cell_section = self._cell_section()
         else:
-            top_line = 'opti conv noautobond fix molmec cartesian\n'
+            # Constant volume.
+            top_line += 'conv '
+            cell_section = ''
+
+        top_line += 'noautobond fix molmec cartesian\n'
 
         position_section = self._position_section(mol, type_translator)
         bond_section = self._bond_section(mol, metal_atoms)
         species_section = self._species_section(type_translator)
 
         library = '\nlibrary uff4mof.lib\n'
 
+        if self._cell is not None:
+            output_cif = output_xyz.replace('xyz', 'cif')
+            periodic_output = f'output cif {output_cif}\n'
+        else:
+            periodic_output = ''
+
         output_section = (
             '\n'
+            f'maxcyc {self._maxcyc}\n'
             'terse inout potentials\n'
-            'terse inout cell\n'
+            'terse in cell\n'
             'terse in structure\n'
             'terse inout derivatives\n'
             f'output xyz {output_xyz}\n'
+            f'{periodic_output}'
             # 'output movie xyz steps_.xyz\n'
         )
 
         with open(in_file, 'w') as f:
             f.write(top_line)
+            f.write(cell_section)
             f.write(position_section)
             f.write(bond_section)
             f.write(species_section)
             f.write(library)
             f.write(output_section)
 
+    def update_cell(self, cif_filename):
```

Comment:
Function name doesn't agree with information. Should be maybe extract_cell, unless it's updating the cell of self. It would also be better if this was a staticmethod, as it doesn't require self

Reply:
Fixed.